api/tests/tooling/database-builder/database-builder.js

```diff
@@ -32,4 +42,38 @@ module.exports = class DatabaseBuilder {
         this.databaseBuffer.purge();
       });
   }
+
+  async listAllTableNames() {
+    const bindings = this.isSQLite ? undefined : [this.databaseName];
+    const resultSet = await this.knex.raw(this.listTablesQuery, bindings);
+
+    const rows = this.isSQLite ? resultSet : resultSet.rows;
+    return _.map(rows, 'table_name');
+  }
+
+  async clearAllTables() {
+    const tableNames = await this.listAllTableNames();
+    const tablesToDelete = _.without(tableNames,
+      'knex_migrations',
+      'knex_migrations_lock',
+      'sqlite_sequence'
+    );
+
+    for(const tableName of tablesToDelete) {
+      await this.knex.raw(this.truncateTableQuery, [tableName]);
+    }
+  }
```

Comment:
Le `truncate` c'est un `clear?` Parce que quand je lis truncate, je pense à troncature, donc à une séparation d'un ensemble. Le clear me parlerait plus ici. On a une fonction `clearAll`, je m'attendrait donc à voir une série de `clear` appelés les uns après les autres

Reply:
Le `truncate table` sert à vider une table, c'est un `delete from` de toutes les lignes optimisé.

> TRUNCATE quickly removes all rows from a set of tables. It has the same effect as an unqualified DELETE on each table, but since it does not actually scan the tables it is faster. Furthermore, it reclaims disk space immediately, rather than requiring a subsequent VACUUM operation. This is most useful on large tables.

in : https://www.postgresql.org/docs/10/sql-truncate.html

Reply:
Avec la notion de `TRUNCATE` il y a aussi la notion de "on conserve la valeur d'index courant de la table ou de la séquence". Ex : si j'ai 12 patates, quand je fais `truncate table patates`, alors je n'ai plus de lignes patates, mais à la prochaine insertion, l'ID de la nouvelle patate sera 13.

Pas de préférence qu'on appelle les méthodes `truncateXxx`, `clearXxx`, `deleteXxx` ou `emptyXxx`.